src/Interpreters/ActionsVisitor.cpp

```diff
@@ -524,11 +524,14 @@ const ActionsDAG & ScopeStack::getLastActions() const
 
 struct CachedColumnName
 {
+    bool & skip_cache;
```

Comment:
A cached column has a flag 'not really cached', and on top of that, it is a reference to somewhere else? This looks broken...